react/features/base/profile/functions.js

```diff
@@ -11,5 +11,21 @@
 export function getProfile(state: Object) {
     const profileStateSlice = state['features/base/profile'];
 
-    return profileStateSlice ? profileStateSlice.profile || {} : {};
+    return _toProfile(profileStateSlice);
+}
+
+/**
+ * Converts the related redux subtree to a profile object, considering
+ * the legacy implementation of the profile too.
+ * FIXME: This may be removed later.
+ *
+ * @param {Object} slice - The Redux state.
+ * @returns {Object}
+ */
+function _toProfile(slice: Object) {
```

Comment:
Considering we don't have a release version with the Settings, is it really necessary to have legacy support for the purposes of an internal release?